src/ivc/public_params.rs

```diff
@@ -203,97 +235,82 @@ where
     RP2: ROPair<C2::Scalar, Config = MainGateConfig<MAIN_GATE_T>>,
 {
     pub fn new(
-        primary: CircuitPublicParamsInput<'key, C1, RP1::Args>,
-        secondary: CircuitPublicParamsInput<'key, C2, RP2::Args>,
+        primary: CircuitPublicParamsInput<'key, '_, A1, C1, RP1::Args, SC1>,
+        secondary: CircuitPublicParamsInput<'key, '_, A2, C2, RP2::Args, SC2>,
         limb_width: NonZeroUsize,
-        limbs_count_limit: NonZeroUsize,
+        limbs_count: NonZeroUsize,
     ) -> Result<Self, Error> {
+        let primary_S = CircuitRunner::new(
+            primary.k_table_size,
+            StepFoldingCircuit::<'_, A1, C2, SC1, RP1::OnCircuit, MAIN_GATE_T> {
+                step_circuit: primary.step_circuit,
+                input: StepInputs::without_witness::<SC1>(
+                    primary.k_table_size,
+                    NUM_IO,
+                    &StepParams::new(limb_width, limbs_count, primary.ro_constant.clone()),
+                ),
+            },
+            vec![C1::Scalar::ZERO, C1::Scalar::ZERO],
+        )
+        .try_collect_plonk_structure()?;
+
+        let secondary_S = CircuitRunner::new(
+            primary.k_table_size,
+            StepFoldingCircuit::<'_, A2, C1, SC2, RP2::OnCircuit, MAIN_GATE_T> {
+                step_circuit: secondary.step_circuit,
+                input: StepInputs::without_witness::<SC2>(
+                    secondary.k_table_size,
+                    NUM_IO,
+                    &StepParams::new(limb_width, limbs_count, secondary.ro_constant.clone()),
+                ),
+            },
+            vec![C2::Scalar::ZERO, C2::Scalar::ZERO],
+        )
+        .try_collect_plonk_structure()?;
+
         debug!("start creating pp");
-        Ok(Self {
+        let mut self_ = Self {
```

Comment:
what convention is to use "self_" instead of "_self"?

Reply:
In Rust if you start variable with `_` it's can be unused & cargo-clippy & cargo-check will not swear. So the standard way to use the keyword for a variable name is `_` at the end of it